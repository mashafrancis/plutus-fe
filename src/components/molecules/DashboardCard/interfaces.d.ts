import { ReactNode } from 'react';

export interface DashboardCardProps {
	heading?: string;
	body: ReactNode;
	redirect?: (cardId: string) => void;
	actionItem?: ReactNode;
	sx?: any;
	[x: string]: any;
}
