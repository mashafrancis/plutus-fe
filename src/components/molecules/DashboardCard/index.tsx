import { Typography, Box } from '@mui/material';
import { DashboardCardProps } from './interfaces';
import { useTheme } from '@mui/material/styles';

const DashboardCard = ({
	heading,
	actionItem,
	body,
	sx,
	...rest
}: DashboardCardProps): JSX.Element => {
	const theme = useTheme();
	return (
		<Box
			sx={{
				height: 'fit-content',
				border: `0.8px solid ${theme.palette.divider}`,
				borderRadius: 2,
				backgroundColor: theme.palette.background.paper,
				marginBottom: 1,
				[theme.breakpoints.up('sm')]: {
					marginRight: 1,
				},
				...sx,
			}}
			{...rest}
		>
			{heading && (
				<Box
					padding={2}
					display={'flex'}
					alignItems="center"
					justifyContent="space-between"
				>
					<Box marginRight={2}>
						<Typography
							sx={{
								fontWeight: 500,
								backgroundColor: 'rgba(25, 103, 210, 0.11)',
								padding: '6px 20px',
								borderRadius: 1,
								width: 'fit-content',
								fontSize: 12,
								[theme.breakpoints.down('sm')]: {
									fontSize: 11,
									paddingLeft: 2,
									paddingRight: 2,
								},
							}}
							variant="subtitle2"
							color="primary"
							data-testid="header"
						>
							{heading}
						</Typography>
					</Box>
					<Box marginLeft={2}>{actionItem}</Box>
				</Box>
			)}
			<Box
				padding={2}
				display={'flex'}
				alignItems="center"
				justifyContent="space-around"
				// sx={{ flex: '1 0 auto', flexDirection: 'row', width: '100%' }}
				data-testid="body"
			>
				{body}
			</Box>
		</Box>
	);
};

export default DashboardCard;
