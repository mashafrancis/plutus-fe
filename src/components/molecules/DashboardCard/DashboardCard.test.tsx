// react libraries
import { render, screen } from '@testing-library/react';

// component
import DashboardCard from './index';

describe('DashboardCard component', () => {
	const props = {
		heading: 'Header',
		body: 'Body',
		redirect: () => jest.fn(),
		classes: '',
	};

	it('should render correctly', () => {
		const { asFragment } = render(<DashboardCard {...props} />);
		expect(asFragment()).toMatchSnapshot();

		const elemHeader = screen.getByTestId('header');
		expect(elemHeader.innerHTML).toBe('Header');

		const elemDetails = screen.getByTestId('body');
		expect(elemDetails.innerHTML).toBe('Body');
	});
});
