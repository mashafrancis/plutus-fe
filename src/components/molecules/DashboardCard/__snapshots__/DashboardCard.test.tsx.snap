// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardCard component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-ra0mqu"
  >
    <div
      class="MuiBox-root css-86nosg"
    >
      <div
        class="MuiBox-root css-1ay9vb9"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-vhi6o1-MuiTypography-root"
          data-testid="header"
        >
          Header
        </h6>
      </div>
      <div
        class="MuiBox-root css-d0uhtl"
      />
    </div>
    <div
      class="MuiBox-root css-1y0pddo"
      data-testid="body"
    >
      Body
    </div>
  </div>
</DocumentFragment>
`;
